api/client/commands.go

```diff
@@ -2159,23 +2159,19 @@ func (cli *DockerCli) createContainer(config *runconfig.Config, hostConfig *runc
 		return nil, err
 	}
 
-	var result engine.Env
-	if err := result.Decode(stream); err != nil {
+	var response *types.ContainerCreateResponse
+	if err := json.NewDecoder(stream).Decode(&response); err != nil {
```

Comment:
You use double pointer tactic here, but not checking for nilness. Looks like potential panic.